minorminer/layout/__init__.py

```diff
@@ -100,23 +119,42 @@ def _parse_layout_parameter(S, T, layout, layout_kwargs):
     """
     # It's two things, one for S and one for T
     if isinstance(layout, tuple):
-        # It's a layout for S
+        # It's a dict layout for S
```

Comment:
This isn't terribly permissive -- as mentioned above with tuples, users might provide strange types that will fail this check whereas the objects support all the operations you expect of those types.

I'd rearrange this:
```
if isinstance(s_layout, Layout):
    S_layout = s_layout
elif callable(s_layout):
    S_layout = s_layout(S, ...)
else:
    # assumes s_layout implements a mapping interface
    S_layout = Layout(S, s_layout, ...)
```